### ISSUE TITLE
LSTMs with XNNPACK: ERROR: Attempting to use a delegate that only supports static-sized tensors with a graph that has dynamic-sized tensors.

### ISSUE THREAD

**Issue by mattiadg** [tensorflow/tensorflow#47504]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source): 2.4.1

**Provide the text output from tflite_convert**

The conversion is actually correct, the problem occurs when trying to use the model with a delegate that requires static-sized tensors. I'm currently testing with XNNPACK Delegate on the tflite benchmark whose output is:
```
STARTING!
Log parameter values verbosely: [0]
Graph: [encoder.tflite]
Enable op profiling: [1]
Use xnnpack: [1]
Loaded model encoder.tflite
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
ERROR: Attempting to use a delegate that only supports static-sized tensors with a graph that has dynamic-sized tensors.
Failed to apply XNNPACK delegate.
Benchmarking failed.
```

**Standalone code to reproduce the issue** 
The problem can be reproduced by using the profiler as following with the model attached:
```
./linux_x86-64_benchmark_model --graph=encoder.tflite --enable_op_profiling=true --use_xnnpack=true
```

https://drive.google.com/file/d/1xZOAS67A0szLJEqdPTC8vtISZXvKcbO6/view?usp=sharing

**Any other info / logs**

I converted an encoder-decoder model for machine translation using two separate tflite models respectively for the encoder and the decoder.

The encoder uses BiLSTM and generates the error in the title. The same error doesn't occur if I use unroll=True, but this is unfeasible for me as I need input sequences of length 100. In this case, I don't get the error but then the usage of RAM explodes.

The decoder, on the other hand, uses input of length 1 and LSTMCell instead of LSTM, and it doesn't produce the error.

Is there a way to make the operation static-length without using unroll?
I have already tried to set the full shape of the input and output shapes of every layer.
Moreover, it is a keras model but I wrap it into a tf.function for which I set the full input shape in the signature, and then I do the conversion from the concrete function but also this doesn't help to get a static size tensor. Using from_saved_model or from_keras_model doesn't help either.

**Comment by abattery** [comment]
@multiverse-tf could you take a look at this?

**Comment by abattery** [comment]
CCing @renjie-liu as well for LSTM.

**Comment by oleid** [comment]
@abattery I'd be interested if this is fixed in more recent versions of tflite.

I've a model with ConvLSTM trained and converted to tflite on tensorflow-2.10.
But I'm running on a device with tensorflow 2.5 and I got exactly the same error message as OP. 
I'm willing to port more recent versions of tflite to the device's toolchain, however, I'm wondering if it's worth it.

**Comment by mohantym** [comment]
Hi @mattiadg !
We are checking to see whether you still need help in this issue . 
I was able to get  below benchmarking result with  TF 2.11 and Android Chipmunk version for your lite model against XNNPack delegate. (NDK 21 was passed in Python configure.py)

`Inference timings in us: Init: 2103, First inference: 105085, Warmup (avg): 96731.2, Inference (avg): 95030.7
`

Could you let us know from your side. 
Thank you!
